### PR TITLE
log4j2 Migration in Forgetme Tool

### PR DESCRIPTION
**Pull request by Yasara-Y**
Log4j1.x has security vulnerability concerns. Hence log4j1.x is migrated to log4j2 with commons-logging and relevant java classes are modified accordingly.

### CODE REVIEW

**Review by vihanga-liyanage — APPROVED** [review]

File: components/org.wso2.carbon.privacy.forgetme.logs/pom.xml [hunk 1]
```diff
@@ -42,6 +38,10 @@
             <artifactId>testng</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>commons-logging</groupId>
+            <artifactId>commons-logging</artifactId>
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.privacy.forgetme.logs/pom.xml:43]: Duplicate dependency

File: components/org.wso2.carbon.privacy.forgetme.logs/src/main/java/org/wso2/carbon/privacy/forgetme/logs/instructions/LogFileInstruction.java [hunk 2]
```diff
@@ -40,7 +40,7 @@
  */
 public class LogFileInstruction implements ForgetMeInstruction {
 
-    private static final Logger log = LoggerFactory.getLogger(LogFileInstruction.class);
+    private static final Log log = LogFactory.getLog(LogFileInstruction.class);
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.privacy.forgetme.logs/src/main/java/org/wso2/carbon/privacy/forgetme/logs/instructions/LogFileInstruction.java:43]: Constants should be in all caps.

File: components/org.wso2.carbon.privacy.forgetme.logs/src/main/java/org/wso2/carbon/privacy/forgetme/logs/instructions/LogFileInstruction.java [hunk 3]
```diff
@@ -61,7 +61,7 @@ public ForgetMeResult execute(UserIdentifier userIdentifier, ProcessorConfig pro
         LogFileProcessor logFileProcessor = new LogFileProcessor();
 
         if (log.isDebugEnabled()) {
-            log.debug("File {} is being processed.", logFile.getName());
+            log.debug("File {} is being processed." + logFile.getName());
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.privacy.forgetme.logs/src/main/java/org/wso2/carbon/privacy/forgetme/logs/instructions/LogFileInstruction.java:64]: Expectation seems to be printing a log line like `File somefile is being processed.`. But now will be printed as `File {} is being processed.somefile`

File: components/org.wso2.carbon.privacy.forgetme.logs/pom.xml (hunk 1 above)
Comment (Yasara-Y) [reply components/org.wso2.carbon.privacy.forgetme.logs/pom.xml:43]: Fixed with https://github.com/wso2/identity-anonymization-tool/pull/141

File: components/org.wso2.carbon.privacy.forgetme.logs/src/main/java/org/wso2/carbon/privacy/forgetme/logs/instructions/LogFileInstruction.java (hunk 2 above)
Comment (Yasara-Y) [reply components/org.wso2.carbon.privacy.forgetme.logs/src/main/java/org/wso2/carbon/privacy/forgetme/logs/instructions/LogFileInstruction.java:43]: Fixed with https://github.com/wso2/identity-anonymization-tool/pull/141

File: components/org.wso2.carbon.privacy.forgetme.logs/src/main/java/org/wso2/carbon/privacy/forgetme/logs/instructions/LogFileInstruction.java (hunk 3 above)
Comment (Yasara-Y) [reply components/org.wso2.carbon.privacy.forgetme.logs/src/main/java/org/wso2/carbon/privacy/forgetme/logs/instructions/LogFileInstruction.java:64]: Fixed with https://github.com/wso2/identity-anonymization-tool/pull/141

**Pull request merged**